test_plans/sub_group_mask.asciidoc

```diff
@@ -16,33 +16,106 @@ is selected on the CTS command line.
 
 === Feature test macro
 
-All of the tests should use `#ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK` so they can be skipped
-if feature is not supported.
+Part of the extension feature is available from revision 1 of the extension and
+tests for it should use `#if SYCL_EXT_ONEAPI_SUB_GROUP_MASK >= 1` and can be
+skipped if the feature is not supported. That part is described in paragraph
+<<member_functions_rev_1>>.
+
+The other part of the feature is available from revision 2 of the extension and
+tests for it should use `#if SYCL_EXT_ONEAPI_SUB_GROUP_MASK >= 2` and can be
+skipped if the feature is not supported. That part is described in paragraph
+<<ctors_rev_2>>.
+
+
 
 == Tests
 
 All of the following tests have these initial steps performed in defferent .cpp files :
 
 * Create a `queue` from the tested device and call `queue::submit()`.
-* Submit a kernel via `handler::parallel_for()` with parameters `sycl::nd_range<1>(range<1>(128), range<1>(32))`
+* Submit a kernel via `handler::parallel_for()` with parameters `sycl::nd_range<1>(range<1>(32), range<1>(32))`
   and lambda finction with parameter `sycl::nd_item<1>`.
+* Lambda function should be decorated with attribute `reqd_sub_group_size(N)` with `N` values = `[8,16,32]`.
+  If device does not support `sub_group` size `N`, test for size `N` can be skipped.
+  `device.get_info<sycl::info::device::sub_group_sizes>()` should be used
+  to check supported `sub_group` sizes.
 * In lambda function use `nd_item::get_sub_group()` to get `sycl::sub_group` instance.
-* Use `sub_group_ballot` to get `sub_group_mask` instance with predicate suitable for test.
 * Use accessor to buffer for type `bool` to save results.
-* For regular test run check is performed only for leader work-item
+* For regular test run check is performed only for leader work-item.
 * For full conformance test run check is performed for all work-items.
 
 === Test for max_bits
 
 Check that member `size_t max_bits` exists.
 
-=== Tests for Member Functions
+=== Tests for constructors and operator= [[ctors_rev_2]]
+
+Features described in this paragraph are available from revision 2 of the
+extension.
+
+==== sub_group_mask() [[default_ctor]]
+
+* Construct `sub_group_mask` instance `s` `with sub_group_mask()` ctor.
+* Check that `s.none() == true`.
+
+==== sub_group_mask(unsigned long long val) [[ull_ctor]]
+
+* Construct `sub_group_mask` instance `s` with `sub_group_mask(unsigned long long val)` ctor
+  and `size_t val`. Where `val` is `item.get_local_linear_id()` and  `0b1010...`.
```

Comment:
Are you saying that each work-item constructs two `sub_group_mask` instances:

```
sb1 = sub_group_mask(item.get_local_linear_id());
sb2 = sub_group_mask(0b1010...);
```

?

Reply:
I think this test is OK, but it would be clearer to explain it this way:

> Run the following test with `VALUE = item.get_local_linear_id()` and with `VALUE = 0b101010...`:
> ```
> size_t val = VALUE;
> sub_group_mask mask{val};
> for (size_t i = 0;  i < mask.size();  i++) {
>   if (mask[id(i)] != ((val >> i) & 1)) {
>     /* test fails */
>   }
> }
> ```

Reply:
Thanks for the clarification! Fixed it.